src/openloco/scenario.cpp

```diff
@@ -23,4 +66,95 @@ namespace openloco::scenario
     {
         call(0x0043C90C);
     }
+
+    // 0x0046E239
+    // NB: called in sub_46E2C0 below, and in date_tick (hook!).
+    void sub_46E239()
+    {
+        for (uint8_t i = 0; i < 32; i++)
+        {
+            uint32_t ebx = currencyMultiplicationFactor[i];
+
+            // Inflating prices?
+            uint64_t edx_eax = _525EDE[i] * _4FDEC0[i];
+            edx_eax >>= 12;
+
+            // Storing two halves of the money struct?
+            _525EDE[i] += static_cast<uint32_t>(edx_eax);
+            currencyMultiplicationFactor[i] += static_cast<uint32_t>(edx_eax >> 32);
+
+            // Inflating prices more?
+            edx_eax = ebx * _4FDEC0[i];
+            edx_eax >>= 12;
+            currencyMultiplicationFactor[i] += static_cast<uint32_t>(edx_eax);
+        }
+
+        call(0x004375F7);
+        ui::WindowManager::invalidate(ui::WindowType::companyList);
+        ui::WindowManager::invalidate(ui::WindowType::buildVehicle);
+        ui::WindowManager::invalidate(ui::WindowType::construction);
+        ui::WindowManager::invalidate(ui::WindowType::terraform);
+        ui::WindowManager::invalidate(ui::WindowType::industryList);
+    }
+
+    // 0x0046E2C0
+    static void sub_46E2C0(uint16_t year)
+    {
+        for (uint8_t i = 0; i < 32; i++)
+        {
+            currencyMultiplicationFactor[i] = 1024;
+            _525EDE[i] = 0;
+        }
+
+        uint32_t ecx = year - 1900;
```

Comment:
If we change this year to below 1900, the game's balance will be thrown off — everything will become more expensive due to 100 extra years worth of inflation.

Perhaps we should branch off here and not compute any inflation prior to 1900, for now?

Reply:
I've implemented the proposed solution, for now.